it/src/test/java/org/corfudb/universe/scenario/RotateLinkFailureIT.java

```diff
@@ -0,0 +1,139 @@
+import org.corfudb.runtime.view.ClusterStatusReport.ClusterStatus;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.universe.GenericIntegrationTest;
+import org.corfudb.universe.group.cluster.CorfuCluster;
+import org.corfudb.universe.node.client.CorfuClient;
+import org.corfudb.universe.node.server.CorfuServer;
+import org.corfudb.util.Sleep;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.corfudb.universe.scenario.ScenarioUtils.waitForLayoutChange;
+import static org.corfudb.universe.scenario.ScenarioUtils.waitForUnresponsiveServersChange;
+import static org.corfudb.universe.scenario.ScenarioUtils.waitUninterruptibly;
+import static org.corfudb.universe.scenario.fixture.Fixtures.TestFixtureConst.DEFAULT_STREAM_NAME;
+import static org.corfudb.universe.scenario.fixture.Fixtures.TestFixtureConst.DEFAULT_TABLE_ITER;
+
+public class RotateLinkFailureIT extends GenericIntegrationTest {
+
+    /**
+     * Test cluster behavior when rotating link failure among nodes
+     * <p>
+     * 1) Deploy and bootstrap a three nodes cluster
+     * 2) Create a link failure between node0 and node1
+     * 3) Create a link failure between node1 and node2
+     *    and heal previous link failure
+     * 4) Create a link failure between node2 and node0
+     *    and heal previous link failure
+     * 5) Reverse rotation direction, create a link failure
+     *    between node1 and node2 and heal previous link failure
+     * 6) Verify layout and data path after each rotation
+     * 7) Recover cluster by removing all link failures
+     * 8) Verify layout, cluster status and data path
+     */
+    @Test(timeout = 300000)
+    public void rotateLinkFailureTest() {
+        getScenario().describe((fixture, testCase) -> {
+            CorfuCluster corfuCluster = universe.getGroup(fixture.getCorfuCluster().getName());
+
+            CorfuClient corfuClient = corfuCluster.getLocalCorfuClient();
+
+            CorfuTable<String, String> table = corfuClient.createDefaultCorfuTable(DEFAULT_STREAM_NAME);
+            for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                table.put(String.valueOf(i), String.valueOf(i));
+            }
+
+            testCase.it("Should rotate link failures among cluster", data -> {
+                CorfuServer server0 = corfuCluster.getServerByIndex(0);
+                CorfuServer server1 = corfuCluster.getServerByIndex(1);
+                CorfuServer server2 = corfuCluster.getServerByIndex(2);
+
+                // 1st link failure rotation, disconnect server0 from server1
+                server0.disconnect(Collections.singletonList(server1));
+                waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
+
+                String firstServerToKick = Collections.max(Arrays.asList(server0.getEndpoint(), server1.getEndpoint()));
```

Comment:
kick is not expressive. maybe toBecomeUnresponsive?

Reply:
I initially wanted to do the similar thing, but it's too long and it's unnecessary in test cases. I think kick is descriptive, since nodes makes decisions and kick other's/themselves out of the responsive set.